docs(release): update docs (#147)
appcelerator/atom-appcelerator-titanium: CONTRIBUTING.md

## Releasing

> To release of the extension the following needs to be followed. Note that the tag must be pushed before the `apm publish` command:

Reviewer: ```suggestion
To release the extension the steps below need to be followed.
```
